Runtime/Properties/SnapZoneProperty.cs

```diff
@@ -62,23 +70,90 @@ public SnapZone SnapZone
         }
 
         private SnapZone snapZone;
-        
+
+        protected virtual void Reset()
+        {
+            if (SnapZone.attachTransform)
+            {
+                return;
+            }
+
+            GameObject attach = new GameObject("Attach Point");
+            attach.transform.SetParent(transform, false);
+            attach.transform.localRotation = Quaternion.identity;
+            attach.transform.localPosition = Vector3.zero;
+            attach.transform.localScale = Vector3.one;
+
+            FieldInfo attachField = typeof(XRBaseInteractor).GetField("m_AttachTransform", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            if (attachField != null)
+            {
+                attachField.SetValue(SnapZone, attach.transform);
+            }
+            else
+            {
+                DestroyImmediate(attach);
+
+                // It seems that the Unity team has renamed that field. Check the SnapZone script. 
+                Debug.Log("We failed to add an attach point for this snap zone property. Please, drop us a word if you see this message.");
+            }
```

Comment:
Maybe add a return here, there is no point to calculate the volume if there is nothing to apply the center, also `attach` is being called after this and could cause an `NRE`.

Reply:
Yes, there should have been a return.